### DISCUSSION TITLE
Bird Elephant chokes on “use class” command

### DISCUSSION THREAD

**Discussion by atari486** [danieldevine/bird-elephant#38] [Q&A] [answered]
Hello, my script is choking on the first statement “use Coderjerk\BirdElephant\BirdElephant;”. I did install composer and require coderjerk/bird-elephant. Here is the error I get:

PHP Fatal error: Uncaught Error: Class “Coderjerk\BirdElephant\BirdElephant” not found in C:\PHP\script.txt:18

I tried installing git and cloning the github repository into various subdirectories within my C:\PHP (Coderjerk, Coderjerk\BirdElephant, etc) but it still cannot find/recognize the class. What should I do? I am a rank beginner in PHP so I am clueless here. If it helps, I am doing this through a command prompt in Windows, not Linux. Thanks.

**Comment by danieldevine** [comment, marked as answer]
Hi,

What way are you autoloading composer dependencies in your project?

**Reply by atari486** [reply]
Hi, thanks for replying. To be honest, I don't even know what that means. I just type in the composer require thing, and then go straight to running my php script. So I guess I'm missing a step. How would I go about autoloading these dependencies you mentioned? I see a number of "composer.json" and "autoload*.php" files in the vendor subdirectory, would any of them lead to victory? Specifically...

1. vendor has "autoload.php" which looks like it just runs autoload_real
2. vendor\composer has a bunch of php's starting with "autoload_": classmap, files, namespaces, psr4, real, static. The psr4 and static ones have some mention of Birdelephant
3. vendor\coderjerk\bird-elephant has a composer.json file. I tried using this to overwrite the one in c:\php and then updating, which installed a truckload of other vendor stuff, but also caused coderjerk\bird-elephant to disappear.

**Reply by danieldevine** [reply]
You need to add 
```php
require __DIR__ . '/vendor/autoload.php';
 ``` 
to the top of your script, assuming its in the same directory as the vendor folder.  This will use composer's autoloader to automatically load the package into your project.  It is going to be very difficult to help you more than that because:
- it sounds like you need to learn some basics before you proceed further with the library.  
- It isn't an issue specific to the library.

If you look at [bootstrap.php](https://github.com/danieldevine/bird-elephant/blob/4fa2cb8ba9549fed137c62e59a74cf87fb89a89e/bootstrap.php), [index.php](https://github.com/danieldevine/bird-elephant/blob/4fa2cb8ba9549fed137c62e59a74cf87fb89a89e/index.php) and [authenticate.php](https://github.com/danieldevine/bird-elephant/blob/4fa2cb8ba9549fed137c62e59a74cf87fb89a89e/authenticate.php) in the root directory of this repo you will see a fully functional example of the library in action. 

It would definitely help you to read more about the topic: 
- https://metabox.io/composer-and-autoload-in-php/
- https://www.merixstudio.com/blog/what-composer-and-how-use-it-project/

Best of luck!
